docs(changelog): layer list updates (#278)
Esri/feedback-js-api-next: CHANGELOG.md

We've updated the properties that control selection and reordering in list widgets. The now deprecated `selectionEnabled` property, which controlled both selection and reordering, has been replaced with `dragEnabled` and `selectionMode`. `dragEnabled` controls reordering, while `selectionMode` manages selection.

`selectionMode` can be set to:
| Value | Description |
| ----- | ----------- |
| multiple | Allows any number of items to be selected at once. This is useful when you want to apply an operation to multiple items at the same time. |
| none | Disables selection. Use this when you want to prevent selecting items. |

> | single | Allows only one item to be selected at a time. If another item is selected, the previous selection is cleared. This is useful when you want to ensure that maximum of one item is selected at a time. |

Reviewer: ```suggestion
| single | Allows only one item to be selected at a time. If another item is selected, the previous selection is cleared. This is useful when you want to ensure that a maximum of one item is selected at a time. |
```

| single-persist | Allows only one item to be selected at a time and prevents de-selection. Once an item is selected, it remains selected until another item is selected. This is useful when you want to ensure that there is always exactly one selected item. |

> By adding `selectionMode`, `multipleSelectionEnabled` is no longer nessessary and it has been depricated and will be removed in a future release.

Reviewer: ```suggestion
By adding `selectionMode`, `multipleSelectionEnabled` is no longer necessary and it has been deprecated and will be removed in a future release.
```

[ListItemPanel](https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-LayerList-ListItemPanel.html)s now do not include any padding by default. We want to leave the spacing inside the ListItemPanel, and its content to the application developer. We have also added a `flowEnabled` property to the ListItemPanels to control if the panel opens in a slot at the bottom of the ListItem or in a [Calcite Flow](https://developers.arcgis.com/calcite-design-system/components/flow/) that allows for drilling in and out of panels.

#### This is a quick summary of the changes.

> - Depricates the `multipleSelectionEnabled` property. Use `selectionMode` instead.
> - Depricates the `selectionEnabled` property. Use `selectionMode` and `dragEnabled` as appropriate.

Reviewer: ```suggestion
- Deprecates the `multipleSelectionEnabled` property. Use `selectionMode` instead.
- Deprecates the `selectionEnabled` property. Use `selectionMode` and `dragEnabled` as appropriate.
```

- Attribution.iconClass deprecated since 4.27. Use icon instead.
- BasemapGallery.iconClass deprecated since 4.27. Use icon instead.
- BasemapLayerList.iconClass deprecated since 4.27. Use icon instead.

> - BasemapLayerList.multipleSelectionEnabled depricated since 4.29. Use selectionMode instead.
> - BasemapLayerList.editingEnabled depricated since 4.29. Use dragEnabled, selectionMode, and visibleElements.editButton instead.

Reviewer: ```suggestion
- BasemapLayerList.multipleSelectionEnabled deprecated since 4.29. Use selectionMode instead.
- BasemapLayerList.editingEnabled deprecated since 4.29. Use dragEnabled, selectionMode, and visibleElements.editButton instead.
```

- InputFieldGroup.visible deprecated since 4.27. Use GroupInput.visible instead.
- InputFieldGroup deprecated since 4.27. Use GroupInput instead.
- LayerList.iconClass deprecated since 4.27. Use icon instead.

> - LayerList.multipleSelectionEnabled depricated since 4.29. Use selectionMode instead.
> - LayerList.selectionEnabled depricated since 4.29. Use dragEnabled and selectionMode instead.

Reviewer: ```suggestion
- LayerList.multipleSelectionEnabled deprecated since 4.29. Use selectionMode instead.
- LayerList.selectionEnabled deprecated since 4.29. Use dragEnabled and selectionMode instead.
```

- SunLighting.ambientOcclusionEnabled deprecated since version 4.27. Ambient occlusion is automatically shown and this property has no effect.
- SunLighting.waterReflectionEnabled deprecated since version 4.27. Reflections are automatically shown and this property has no effect.
- Swipe.iconClass deprecated since 4.27. Use icon instead.

> - TableList.multipleSelectionEnabled depricated since 4.29. Use selectionMode instead.
> - TableList.selectionEnabled depricated since 4.29. Use dragEnabled and selectionMode instead.

Reviewer: ```suggestion
- TableList.multipleSelectionEnabled deprecated since 4.29. Use selectionMode instead.
- TableList.selectionEnabled deprecated since 4.29. Use dragEnabled and selectionMode instead.
```
